src/web/extractors.rs

```diff
@@ -1023,6 +1023,22 @@ impl FromRequest for Box<dyn Db> {
     }
 }
 
+#[derive(Debug, Default, Clone, Deserialize, Validate)]
+#[serde(default)]
+pub struct Offset {
+    pub timestamp: Option<SyncTimestamp>,
+    pub offset: i64,
```

Comment:
Can this be u64?

Reply:
I did try that, but there's a lot of existing code that uses -1 as a sentinel, and fixing all the typeerrors seems like a whole lot of work. If you really think it is important, I will do it.

Reply:
Might be worth doing as a separate issue from this one

Reply:
Filed https://github.com/mozilla-services/syncstorage-rs/issues/414